kaldb/src/main/java/com/slack/kaldb/server/RecoveryTaskCreator.java

```diff
@@ -103,8 +103,8 @@ public static long getHighestDurableOffsetForPartition(
     return Math.max(maxRecoveryOffset, maxSnapshotOffset);
   }
 
-  private static String getRecoveryTaskName(long creationTimeEpochMs, String partitionId) {
-    return "recoveryTask_" + partitionId + "_" + creationTimeEpochMs;
+  private static String getRecoveryTaskName(String partitionId) {
+    return "recoveryTask_" + partitionId + "_" + Instant.now().getNano();
```

Comment:
Is there a reason for using a nano here? We can't have contention trying to create a recovery task for the same partition ID, and this makes the znode path much longer than is needed.

Reply:
I used millis initially and ended up with some race conditions in unit tests so I switched to nano.  It's about 13 chars as oppsed to 10 so it was not too bad in terms of string length. So kept it as is.